internal/tool/kubectl/runner.go

```diff
@@ -152,8 +152,18 @@ func (r *Runner) Delete(manifestPath string, params ...string) error {
 }
 
 func (r *Runner) Version() (string, error) {
+	args := []string{"version"}
+
+	if r.paths.Kubeconfig != "" {
+		args = append(args, "--kubeconfig", r.paths.Kubeconfig)
+	}
+
+	if !r.serverSide {
+		args = append(args, "--client")
```

Comment:
Correct me if I'm wrong, but this logic is not correct. `serverSide` is used for example, when you do `kubectl apply --server-side`, but it is not correct to assume that if I want to do a server side apply I will also want to run `kubectl version` without the `--client` flag.

The two flags are not related.

I'd suggest separating the properties.

Reply:
_haste makes waste_